JOSESwift/Sources/PBES2KeyEncryptionMode.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  PBES2KeyEncryptionMode.swift
+//  JOSESwift
+//
+//  Created by Tobias Hagemann on 08.12.23.
+//
+//  ---------------------------------------------------------------------------
+//  Copyright 2023 Airside Mobile Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  ---------------------------------------------------------------------------
+//
+
+import Foundation
+
+struct PBES2KeyEncryptionMode {
+    typealias KeyType = String
+
+    private let keyManagementAlgorithm: KeyManagementAlgorithm
+    private let contentEncryptionAlgorithm: ContentEncryptionAlgorithm
+    private let password: KeyType
+
+    init(
+        keyManagementAlgorithm: KeyManagementAlgorithm,
+        contentEncryptionAlgorithm: ContentEncryptionAlgorithm,
+        password: KeyType
+    ) {
+        self.keyManagementAlgorithm = keyManagementAlgorithm
+        self.contentEncryptionAlgorithm = contentEncryptionAlgorithm
+        self.password = password
+    }
+}
+
+extension PBES2KeyEncryptionMode: EncryptionKeyManagementMode {
+    func determineContentEncryptionKey(for header: JWEHeader) throws -> Data {
+        var updatedHeader = header
+        let saltInput = try SecureRandom.generate(count: 16)
```

Comment:
The [recommended minimum is 8 octets](https://datatracker.ietf.org/doc/html/rfc7518#section-4.8.1.1). 16 feels a bit arbitrary, should we switch this to 8 (and ideally give the user the choice to specify this length themselves, see my other comment).

Reply:
I guess, there is no right or wrong here. Giving the user the choice is probably best. Not sure if 16 is still a better "default", see discussions like: https://security.stackexchange.com/questions/110084/parameters-for-pbkdf2-for-password-hashing

Reply:
You're right. I went with 8 as default (as it is the only reference to a number I found in the RFC) but opened it up for the user to specify whatever they want (as long as it's over 8 of course). Let me know what you think.

Reply:
Sounds great! 👍